dao/searchDao.js

```diff
@@ -1,21 +1,15 @@
 const db = require('./db');
-let sqliteDatabase;
-if(process.env.NODE_ENV === 'test') {sqliteDatabase = require('better-sqlite3');}
-
+const allFilters = require('./allFilters');
+let sqliteDatabase = (process.env.NODE_ENV === 'test') ? require('better-sqlite3') : null;
 exports.getAll = function(filter, cb) {
-    let classFilter = '';
-    if(filter.class) {
-        classFilter = ' AND class = ' + db.escape(filter.class);
-    }
-    let sqlQuery = 'SELECT * FROM tbl_tickets WHERE reserver IS NULL' + classFilter + ';';
-
-    if(process.env.NODE_ENV === 'test') {
-        // Please look at the README for debugging to the console
-        const Testdb = new sqliteDatabase('airlinesTest.db', {memory: true});
-        const result = Testdb.prepare(sqlQuery).all();
+    let sqlQuery = 'SELECT flight, seat_row, seat, class, reserver, price, reservation_timeout, booking_id, departure, destination, departure_date, arrival_date, flight_number FROM tbl_tickets AS t LEFT JOIN tbl_flights AS f ON t.flight = f.id WHERE reserver IS NULL' +
+        allFilters.classFilter(filter.class, db) + allFilters.seatFilter(filter.seat, db) +
+        allFilters.departureDateFilter(filter.departureDateAfter, filter.departureDateBefore, db) + allFilters.arrivalDateFilter(filter.arrivalDateAfter, filter.arrivalDateBefore, db) +
+        allFilters.departureLocationFilter(filter.departure_location, db) + allFilters.destinationLocationFilter(filter.destination_location, db) + ';';
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
It sure looks to me like you're creating a new DB connection (and a new SQLite database in the test environment, unless the `better-sqlite3` package reuses the same database) every time the `getAll` method is called. This should not be.

Reply:
However, I never use the connection itself. I used the function that it provides. As far as I know, it uses the same database.

Reply:
You should get the connection to the database *once*, then use "the function it provides" each time.